vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -405,6 +409,80 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
+
+                      // push
+                      log.info("pushing /" + dependency.getFilePath());
+                      writeCacheHeaders(request, filePropsAsyncResult.result());
+                      response.push(HttpMethod.GET, "/" + dependency.getFilePath(), pushAsyncResult -> {
+                        if (pushAsyncResult.succeeded()) {
+                          HttpServerResponse res = pushAsyncResult.result();
+                          final String depContentType = MimeMapping.getMimeTypeForExtension(file);
+                          if (depContentType != null) {
+                            if (depContentType.startsWith("text")) {
+                              res.putHeader("Content-Type", contentType + ";charset=" + defaultContentEncoding);
+                            } else {
+                              res.putHeader("Content-Type", contentType);
+                            }
+                          }
+                        }
+                      });
+                    }
+                  });
+                }
+              }
+            }
+          } else if (http2PushMap != null) {
+            //Link preload when file push is not supported
+
+            for (Map.Entry<String, List<Http2PushMapping>> http2PushEntry : http2PushMap.entrySet()) {
+
+              List<String> asLinkExtensions = new ArrayList<>();
+              HttpServerResponse response = request.response();
+
+              for (Http2PushMapping dependency : http2PushEntry.getValue()) {
+                if(!dependency.isNoPush()) {
+                  final String dep = webRoot + "/" + dependency.getFilePath();
+
+                  // get the file props
+                  log.info("looking up props for " + dep);
+                  getFileProps(context, dep, filePropsAsyncResult -> {
+                    if (filePropsAsyncResult.succeeded()) {
+                      // push
+                      log.info("pushing /" + dependency.getFilePath());
+                      writeCacheHeaders(request, filePropsAsyncResult.result());
+                      final String depContentType = MimeMapping.getMimeTypeForExtension(dependency.getFilePath());
+                      String asLinkExtension = "<" + dependency.getFilePath() + ">; rel=preload; as=" + dependency.getExtensionTarget();
+                      asLinkExtensions.add(asLinkExtension);
+
+                      if (depContentType != null) {
+                        if (depContentType.startsWith("text")) {
+                          response.putHeader("Content-Type", contentType + ";charset=" + defaultContentEncoding);
+                        } else {
+                          response.putHeader("Content-Type", contentType);
+                        }
+                      }
+                    }
+                  });
+                }
+              }
+              response.putHeader("Link", String.join(",", asLinkExtensions));
```

Comment:
There should be a link header per push, see: https://w3c.github.io/preload/#server-push-(http/2)

example:

```
Link: <https://example.com/font.woff2>; rel=preload; as=font; type="font/woff2"
```